test/integration/runner/runner_test.go

```diff
@@ -0,0 +1,140 @@
+package runner
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"testing"
+)
+
+var timeout = flag.Duration("sub.timeout", 0, "Specify the timeout for each sub test")
+
+func TestIntegration(t *testing.T) {
+	dir := ".."
+	packageName := "github.com/openshift/origin/test/integration"
```

Comment:
If I'm reading all this correctly, I don't see an obvious way to make this work for multiple sub-directories of tests.  We'd end up running each bucket serially with respect to other buckets.  The tests in an individual bucket would run in parallel, but each bucket is serial.

Is there a way to avoid that?